client/src/components/Add-Edit-Form/index.js

```diff
@@ -1,14 +1,108 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
 import PropTypes from 'prop-types';
-import { Form, Input, Button, message } from 'antd';
+import { Form, Input, Button, message, Spin } from 'antd';
 import './style.css';
 import AdminContainer from '../AdminContainer';
 
 const axios = require('axios');
 
-const AddEditForm = ({ formType, addLink, editLink, cohortId, inputData }) => {
-  const onFinish = async (values) => {
+const redirectFunc = (push, formType, cohortId, projectType) => {
+  if (formType === 'cohort') {
+    push('/admin/cohorts');
+  } else if (formType === 'student') {
+    push(`/admin/cohorts/${cohortId}/students`);
+  } else {
+    push(`/admin/cohorts/${cohortId}/projects?type=${projectType}`);
+  }
+};
+
+class AddEditForm extends Component {
+  state = {
+    myData: {},
+    // eslint-disable-next-line react/destructuring-assignment
+    addOrEdit: this.props.addLink,
+    projectType: '',
+  };
+
+  async componentDidMount() {
+    const {
+      history: { push },
+      editLink,
+      formType,
+      match: {
+        params: { cohortId, studentId, projectId },
+      },
+    } = this.props;
+
+    try {
+      if (editLink) {
+        if (formType === 'cohort') {
```

Comment:
@MohammedAlghazali it might help code readability if these three if-else blocks were moved into functions.

```
if (formType === 'cohort') {
  this.getCohort();
} else if (formType === 'students') {
  this.getStudents();
} else {
  this.getProject();
}
```

Reply:
yes, it will be much better